packages/react/test/specs/components/Dropdown/Dropdown-test.tsx

```diff
@@ -34,20 +35,169 @@ describe('Dropdown', () => {
     })
   })
 
+  describe('open', () => {
+    const onOpenChange = jest.fn()
+
+    afterEach(() => {
+      onOpenChange.mockReset()
+    })
+
+    it('is false when closed by trigger button click', () => {
+      const wrapper = mountWithProvider(<Dropdown onOpenChange={onOpenChange} items={items} />)
+      const triggerButton = wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`)
+
+      triggerButton.simulate('click').simulate('click')
+
+      expect(onOpenChange).toBeCalledTimes(2)
+      expect(onOpenChange).toHaveBeenLastCalledWith(
+        null,
+        expect.objectContaining({
+          open: false,
+        }),
+      )
+    })
+
+    it('is false when closed by hitting Escape in search input', () => {
+      const wrapper = mountWithProvider(
+        <Dropdown onOpenChange={onOpenChange} search items={items} />,
+      )
+      const searchInput = wrapper.find(`input.${DropdownSearchInput.slotClassNames.input}`)
+
+      searchInput
+        .simulate('click')
+        .simulate('change', { target: { value: 'test' } })
+        .simulate('keydown', { keyCode: keyboardKey.Escape, key: 'Escape' })
```

Comment:
_Not related to this PR:_
It's strange to me that we should pass `key` always for Downshift: https://github.com/downshift-js/downshift/blob/8f987928e6a0ff4d0d9e6259748bcd243532f0ec/src/utils.js#L252

It will be great to reuse `keyboard-key` there an simplify it...

Reply:
I will try to take a look in a future PR.